checkstyle/sun_checks.xml

```diff
@@ -201,7 +201,18 @@
     <property name="format" value="master"/>
     <property name="ignoreCase" value="true"/>
     <property name="message" value="Usage should be switched to cluster manager" />
-    <property name="severity" value="ignore"/>
+    <property name="severity" value="error"/>
+  </module>
+
+  <module name="SuppressWithPlainTextCommentFilter">
+    <property name="offCommentFormat" value="CS-SUPPRESS-ALL: .+"/> <!-- Require an explaination after surpressing -->
```

Comment:
Why do this regex and the one below not match?

Should these both be `([\w\|]+) .+`?

Reply:
`CS-SUPPRESS-SINGLE` - suppresses a single named checkstyle rule, then afterward a reason should still be included.   This is useful for disabling a single unavoidable violation, such as where we have upstream libraries with non-inclusive terms in a class name, see the usage in MigrateApiAction.java.  This keeps rules enforced on all other checkstyle checks, such as AvoidStarImport.

`CS-SUPPRESS-ALL` - suppresses _all_ checkstyle rules.  This would be more for a scenario where we need to import a file from another codebase that has entirely different rules, and applying our own rules would be onerous or make merging future changes a nightmare.

#### Example usage where `RegexpSingleline` is suppressed and other checks are still enforced.
https://github.com/opensearch-project/security/blob/f1e804489c82c60442b84dee34e8fa603a4cb636/src/main/java/org/opensearch/security/dlic/rest/api/MigrateApiAction.java#L14-L73